test/e2e/specs/editor/various/datepicker.spec.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+// Set browser to a timezone that's different to `timezone`.
+test.use( {
+	timezoneId: 'America/New_York',
+} );
+
+// The `timezone` setting exposed via REST API only accepts `UTC`
+// and timezone strings by location.
+const TIMEZONES = [ 'Pacific/Honolulu', 'UTC', 'Australia/Sydney' ];
+
+TIMEZONES.forEach( ( timezone ) => {
+	test.describe( `Datepicker: ${ timezone }`, () => {
+		let orignalTimezone;
+		test.beforeAll( async ( { requestUtils } ) => {
+			orignalTimezone = ( await requestUtils.getSiteSettings() ).timezone;
+			await requestUtils.updateSiteSettings( { timezone } );
+		} );
+
+		test.beforeEach( async ( { admin, editor } ) => {
+			await admin.createNewPost();
+			await editor.openDocumentSettingsSidebar();
+		} );
+
+		test.afterAll( async ( { requestUtils } ) => {
+			await requestUtils.updateSiteSettings( {
+				timezone: orignalTimezone,
+			} );
+		} );
+
+		test( 'should show the publishing date as "Immediately" if the date is not altered', async ( {
+			page,
+		} ) => {
+			await expect(
+				page.getByRole( 'button', { name: 'Change date' } )
+			).toHaveText( 'Immediately' );
+		} );
+
+		test( 'should show the publishing date if the date is in the past', async ( {
+			page,
+		} ) => {
+			const datepicker = page.getByRole( 'button', {
+				name: 'Change date',
+			} );
+			await datepicker.click();
+
+			// Change the publishing date to a year in the future.
+			await page
+				.getByRole( 'group', { name: 'Date' } )
+				.getByRole( 'spinbutton', { name: 'Year' } )
+				.click();
+			await page.keyboard.press( 'ArrowDown' );
+			await page.keyboard.press( 'Escape' );
+
+			// The expected date format will be "Sep 26, 2018 11:52 pm".
+			await expect(
+				page.getByRole( 'button', { name: 'Change date' } )
+			).toContainText( /^[A-Za-z]+\s\d{1,2},\s\d{1,4}/ );
```

Comment:
@kevin940726, @WunderBart, what do you think about this assertion? Would it suffice? 😅

Reply:
I think it might be easier to just parse the date 😆. Something like...

```js
const dateString = await page.getByRole( 'button', { name: 'Change date' } ).textContent();
const date = new Date( dateString );
expect( date ).toEqual( datePickerDate );
```

We can keep the regex to test the format once if we want to.

Reply:
Thanks for the example; I'll try it out later.

P.S. My understanding of this test is that they just assert that the date label switches from `Immediately` to an actual date. In my opinion, it shouldn't really matter if the date string matches.

Reply:
Tried the suggested path, but `Date` fails to parse the string returned by a locator.

Reply:
Judged from the test case, I think it's also asserting that the date in the datepicker is the same as the one shown on the button. Not sure if it matters though!

Reply:
I believe unit tests cover exact date assertions for the `date-time` components. Doing something similar usually ends up in ugly e2e test code 😅

Should we land the current migration version or try to make date assertions happen?

Reply:
If it's already covered by unit tests then absolutely! TBH, I think all of these cases can be tested in unit/integration tests 😆. I was just being conservative 😅.